packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -79,85 +84,105 @@ def get_metrics_prom(dcos_api_session, node):
 
 def test_metrics_agents_mesos(dcos_api_session):
     """Assert that mesos metrics on agents are present."""
-    for agent in dcos_api_session.slaves:
+    if dcos_api_session.slaves:
         @retrying.retry(wait_fixed=STD_INTERVAL, stop_max_delay=METRICS_WAITTIME)
         def check_mesos_metrics():
-            response = get_metrics_prom(dcos_api_session, agent)
+            response = get_metrics_prom(dcos_api_session, dcos_api_session.slaves[0])
+            assert 'mesos_slave_uptime_secs' in response.text
+        check_mesos_metrics()
+
+    if dcos_api_session.public_slaves:
+        @retrying.retry(wait_fixed=STD_INTERVAL, stop_max_delay=METRICS_WAITTIME)
+        def check_mesos_metrics():
+            response = get_metrics_prom(dcos_api_session, dcos_api_session.public_slaves[0])
```

Comment:
The bodies of these two `if` statements are identical. This test could instead collect the agents we intend to test into a list, and then run through that list in a `for` loop.

```
nodes = []
if dcos_api_session.slaves:
    nodes.append(dcos_api_session.slaves[0])
if dcos_api_session.public_slaves:
    nodes.append(dcos_api_session.public_slaves[0])

for node in nodes:
    @retrying.retry(wait_fixed=STD_INTERVAL, stop_max_delay=METRICS_WAITTIME)
    def check_mesos_metrics():
        response = get_metrics_prom(dcos_api_session, node)
        assert 'mesos_slave_uptime_secs' in response.text
    check_mesos_metrics()
```

And this is doing the same exact thing as `test_metrics_masters_mesos()`. We could test a master here as well.